File: SIN BDD/SOAP_JAVA_CONUNI_GR06/08. SERVIDOR/servidor_soap_java_conuni_gr06/test/ec/edu/monster/prueba/pruebaLongitud.java
package ec.edu.monster.prueba;

import ec.edu.monster.servicio.ServicioLongitud;
import org.junit.Test;
import static org.junit.Assert.*;

public class pruebaLongitud {

    private final ServicioLongitud servicio = new ServicioLongitud();
    private static final double MARGEN = 0.0001;

    @Test
    public void pruebaMetrosAPies() {
        assertEquals(3.28084, servicio.metrosAPies(1.0), MARGEN);
    }

    @Test
    public void pruebaKilometrosAMillas() {
        assertEquals(6.21371, servicio.kilometrosAMillas(10.0), MARGEN);
    }

    @Test
    public void pruebaCentimetrosAPulgadas() {
        assertEquals(1.0, servicio.centimetrosAPulgadas(2.54), MARGEN);
    }

    @Test
    public void pruebaYardasAMetros() {
        assertEquals(1.0, servicio.yardasAMetros(1.09361), MARGEN);
    }

    @Test
    public void pruebaMilimetrosAPulgadas() {
        assertEquals(0.393701, servicio.milimetrosAPulgadas(10.0), MARGEN);
    }

    @Test
    public void pruebaValorCero() {
        assertEquals(0.0, servicio.metrosAPies(0.0), MARGEN);
    }
}

File: SIN BDD/SOAP_DOTNET_CONUNI_GR06/08. SERVIDOR/servidor_soap_dotnet_conuni_gr06/Pruebas/PruebaLongitud.cs
using Ec.Edu.Monster.Controlador;

namespace Ec.Edu.Monster.Pruebas;

static class PruebaLongitud
{
    static readonly CONUNI servicio = new();
    static int ok = 0, fail = 0;

    public static void Ejecutar()
    {
        Console.WriteLine("=== PRUEBAS DE CONVERSION DE LONGITUD ===");
        Console.WriteLine();

        Probar("metrosAPies", 10, 32.8084);
        Probar("kilometrosAMillas", 5, 3.106855);
        Probar("centimetrosAPulgadas", 10, 3.93701);
        Probar("yardasAMetros", 10, 9.144);
        Probar("milimetrosAPulgadas", 25.4, 1.0);

        Console.WriteLine();
        Console.WriteLine($"--- RESULTADOS LONGITUD: OK={ok} FAIL={fail} ---");
        Console.WriteLine();
    }

    static void Probar(string operacion, double valor, double esperado)
    {
        try
        {
            double obtenido = operacion switch
            {
                "metrosAPies" => servicio.MetrosAPies(valor),
                "kilometrosAMillas" => servicio.KilometrosAMillas(valor),
                "centimetrosAPulgadas" => servicio.CentimetrosAPulgadas(valor),
                "yardasAMetros" => servicio.YardasAMetros(valor),
                "milimetrosAPulgadas" => servicio.MilimetrosAPulgadas(valor),
                _ => throw new ArgumentOutOfRangeException(nameof(operacion))
            };

            bool casoOk = Math.Abs(obtenido - esperado) <= ConstantesPruebas.Tolerancia;

            if (casoOk)
            {
                ok++;
                Console.WriteLine($"[OK] {operacion}({valor}) = {obtenido:F4}");
            }
            else
            {
                fail++;
                Console.WriteLine($"[FAIL] {operacion}({valor})");
                Console.WriteLine($"  Esperado: {esperado} +/- {ConstantesPruebas.Tolerancia}");
                Console.WriteLine($"  Obtenido: {obtenido}");
            }
        }
        catch (Exception e)
        {
            fail++;
            Console.WriteLine($"[FAIL] {operacion}({valor}) -> Error: {e.Message}");
        }
    }
}
